frontend/src/core/comments/components/AddComment.js

```diff
@@ -1,97 +1,393 @@
         }
-        else {
-            commentInput.current.rows = maxRows;
+        if (teamsScroll) {
+            teamsScroll.addEventListener('scroll', handleScroll);
+        }
+
+        return () => {
+            if (historyScroll) {
+                historyScroll.removeEventListener('scroll', handleScroll);
+            }
+            if (teamsScroll) {
+                teamsScroll.removeEventListener('scroll', handleScroll);
+            }
+        }
+    }, [])
+
+    const handleMentionsKeyDown = React.useCallback(
+        (event) => {
+            if (!target) { 
+                return;
+            }
+            switch (event.key) {
+                case 'ArrowDown': {
+                    event.preventDefault();
+                    const prevIndex = index >= chars.length - 1 ? 0 : index + 1;
+                    setIndex(prevIndex);
+                    break;
+                }
+                case 'ArrowUp': {
+                    event.preventDefault();
+                    const nextIndex = index <= 0 ? chars.length - 1 : index - 1;
+                    setIndex(nextIndex);
+                    break;
+                }
+                case 'Tab':
+                case 'Enter':
+                    event.preventDefault();
+                    Transforms.select(editor, target);
+                    insertMention(editor, chars[index], usersList);
+                    setTarget(null);
+                    break;
+                case 'Escape':
+                    event.preventDefault();
+                    setTarget(null);
+                    break;
+                default:
+                    return;
+            }
+        },
+        [ chars, editor, index, target, usersList ]
+    );
+
+    const handleEditorKeyDown = (event: SyntheticKeyboardEvent<>) => {
+        if (event.key === 'Enter' && !event.shiftKey) {
+            event.preventDefault();
+            submitComment();
+        }
+        if (event.key === 'Enter' && event.shiftKey) { 
+            event.preventDefault();
+            /*
+            * This allows for the new lines to render while adding comments.
+            * To avoid an issue with the cursor placement and an error when 
+            * navigating with arrows that occurs in Firefox '\n' can't be 
+            * the last character so the BOM was added
+            */
+            editor.insertText('\n\uFEFF');
+            return null;
         }
     }
+    
+    const handleMentionsMouseDown = React.useCallback((event: SyntheticMouseEvent<HTMLDivElement>) => {
+        event.preventDefault();
+        if (target !== null) {
+            const charIndex = chars.indexOf(event.currentTarget.innerText)
+            Transforms.select(editor, target);
+            insertMention(editor, chars[charIndex], usersList);
+            return setTarget(null);
+        }
+    }, [ editor, target, chars, usersList ]);
+    
+    const getUserGravatar = React.useCallback((name: string) => {
```

Comment:
This one on the other hand I think is smart, because it would only change when `usersList` changes, and should never happen more than once (on the first load of the component, with the current implementation, and even less if you move that loading to the `App` component).